File: nuget/src/ReactUbiquitous.NuGet/Builders/PageBuilder.cs
using ReactUbiquitous.NuGet.Models;
using ReactUbiquitous.NuGet.Models.Sections;

namespace ReactUbiquitous.NuGet.Builders;

/// <summary>Provides a fluent builder for constructing <see cref="UIPageConfig"/> instances.</summary>
public class PageBuilder
{
    private readonly UIPageConfig _page = new();

    /// <summary>Sets the unique page identifier.</summary>
    public PageBuilder WithId(string id) { _page.Id = id; return this; }

    /// <summary>Sets the page title.</summary>
    public PageBuilder WithTitle(string title) { _page.Title = title; return this; }

    /// <summary>Sets the page description.</summary>
    public PageBuilder WithDescription(string description) { _page.Description = description; return this; }

    /// <summary>Sets the icon associated with the page.</summary>
    public PageBuilder WithIcon(string icon) { _page.Icon = icon; return this; }

    /// <summary>Sets the display order of the page.</summary>
    public PageBuilder WithOrder(int order) { _page.Order = order; return this; }

    /// <summary>Sets additional CSS class names for the page.</summary>
    public PageBuilder WithClassName(string className) { _page.ClassName = className; return this; }

    /// <summary>Sets whether the page is visible.</summary>
    public PageBuilder WithVisible(bool visible) { _page.Visible = visible; return this; }

    /// <summary>Adds a section to the page.</summary>
    public PageBuilder AddSection(BaseSectionConfig section) { (_page.Sections ??= []).Add(section); return this; }

    /// <summary>Builds and returns the configured page.</summary>
    public UIPageConfig Build() => _page;
}

File: src/Builders/PageBuilder.php
<?php

namespace ReactUbiquitous\Builders;

use ReactUbiquitous\DTOs\UIPageConfig;
use ReactUbiquitous\Contracts\SectionConfigInterface;

final class PageBuilder
{
    private array $sections = [];
    private ?string $description = null;
    private ?string $icon = null;
    private ?string $className = null;
    private ?bool $visible = null;

    public function __construct(
        private readonly string $id,
        private readonly string $title,
        private readonly int $order,
    ) {}

    public static function make(string $id, string $title, int $order = 1): self
    {
        return new self($id, $title, $order);
    }

    public function description(string $description): self
    {
        $clone = clone $this;
        $clone->description = $description;
        return $clone;
    }

    public function icon(string $icon): self
    {
        $clone = clone $this;
        $clone->icon = $icon;
        return $clone;
    }

    public function className(string $className): self
    {
        $clone = clone $this;
        $clone->className = $className;
        return $clone;
    }

    public function visible(bool $visible): self
    {
        $clone = clone $this;
        $clone->visible = $visible;
        return $clone;
    }

    public function addSection(SectionConfigInterface $section): self
    {
        $clone = clone $this;
        $clone->sections[] = $section;
        return $clone;
    }

    public function sections(array $sections): self
    {
        $clone = clone $this;
        $clone->sections = $sections;
        return $clone;
    }

    public function build(): UIPageConfig
    {
        return new UIPageConfig(
            id: $this->id,
            title: $this->title,
            order: $this->order,
            sections: $this->sections,
            description: $this->description,
            icon: $this->icon,
            className: $this->className,
            visible: $this->visible,
        );
    }
}
